src/runtime/streaming-importer.py

```
@@ -0,0 +1,995 @@
+    
+    async def memory_cleanup(self) -> None:
+        """Perform memory cleanup."""
+        collected = gc.collect()
+        if MALLOC_TRIM_AVAILABLE:
+            malloc_trim(0)
+        logger.debug(f"Memory cleanup: collected {collected} objects")
+    
+    def get_collection_name(self, project_path: str) -> str:
+        """Get collection name for project."""
+        normalized = normalize_project_name(project_path)
+        project_hash = hashlib.md5(normalized.encode()).hexdigest()[:8]
```

Ruff (0.13.3)

Probable use of insecure hash functions in `hashlib`: `md5`

(S324)